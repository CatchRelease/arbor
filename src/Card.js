import styled from '@emotion/styled';
import Box from './Box';

const Card = styled(Box.withComponent('div'))`
  border-radius: ${props => props.theme.borderRadius.large};
  border: 1px solid ${props => props.theme.colors.grey20};
  width: 100%;
`;

export default Card;
