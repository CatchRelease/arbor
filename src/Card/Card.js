import styled from '@emotion/styled';
import Box from '../Box';

const Card = styled(Box.withComponent('div'))`
  border-radius: ${props => props.theme.borderRadius.large};
  border: 1px solid ${props => props.theme.colors.border.default};
  width: 100%;
`;

Card.defaultProps = {
  boxShadow: 'elevation0'
};

export default Card;
