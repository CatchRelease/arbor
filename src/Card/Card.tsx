import { FC } from 'react';
import Pane, { PaneProps } from '../Pane';

export type Props = PaneProps;

const Card: FC<Props> = (props) => <Pane {...props} />;

Card.defaultProps = {
  borderRadius: 'large'
};

export default Card;
