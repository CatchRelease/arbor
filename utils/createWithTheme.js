import React from 'react';
import { CacheProvider } from '@emotion/core';
import createCache from '@emotion/cache';
import { create } from 'react-test-renderer';
import { ThemeProvider } from 'emotion-theming';
import theme from '../src/theme';

const myCache = createCache();
myCache.compat = true;

export default (Component) =>
  create(
    <CacheProvider value={myCache}>
      <ThemeProvider theme={theme}>{Component}</ThemeProvider>
    </CacheProvider>
  );
