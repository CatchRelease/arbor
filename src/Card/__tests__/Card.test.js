import React from 'react';
import createWithTheme from '../../../tests/utils/createWithTheme';

import Card from '../Card';

describe('<Card />', () => {
  it('renders a Card properly', () => {
    const tree = createWithTheme(<Card />);

    expect(tree).toMatchSnapshot();
  });
});
