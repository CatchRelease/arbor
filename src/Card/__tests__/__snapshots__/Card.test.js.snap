// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders a Card properly 1`] = `
.emotion-0 {
  font-family: Motiva Sans;
  background-color: #FFFFFF;
  overflow: hidden;
  box-sizing: border-box;
  border-radius: 6px;
  box-shadow: 0 0 1px rgba(37,39,45,0.47);
  display: grid;
  width: 100%;
}

<div
  className="emotion-0 emotion-1"
  display="grid"
  overflow="hidden"
  width="100%"
/>
`;
